IBPSA/Airflow/Multizone/BaseClasses/flowElementData.mo

```diff
@@ -0,0 +1,55 @@
+within IBPSA.Airflow.Multizone.BaseClasses;
+function flowElementData
+  "CubicSplineData with last points linearly interpolated, yd values must be increasing"
+
+  input Real u "independent variable";
+  input Real table[:,:];
+
+  output Real z "Dependent variable with monotone interpolation";
+
+protected
+  Real[:] xd=table[:,1] "Support points";
+  Real[size(xd, 1)] yd=table[:,2] "Support points";
+  Real[size(xd, 1)] d(each fixed=false);
+
+  Integer i "Integer to select data interval";
+
+algorithm
+
+    // Get the derivative values at the support points, monotonicity is ensured
+
+  d := IDEAS.Utilities.Math.Functions.splineDerivatives(
+    x=xd,
+    y=yd,
+    ensureMonotonicity=true);
+
+  i := 1;
+  for j in 1:size(xd, 1) - 1 loop
+    if u > xd[j] then
+      i := j;
+    end if;
+  end for;
+
+  // Extrapolate or interpolate the data
+
+  if i == (size(xd, 1) - 1) then
+    z:=yd[end-1]+(u-xd[end-1])*(yd[end]-yd[end-1])/(xd[end]-xd[end-1]);
+  else
+    z :=IDEAS.Utilities.Math.Functions.cubicHermiteLinearExtrapolation(
+        x=u,
+        x1=xd[i],
+        x2=xd[i + 1],
+        y1=yd[i],
+        y2=yd[i + 1],
+        y1d=d[i],
+        y2d=d[i + 1]);
+  end if;
+
+  annotation (
+    Icon(coordinateSystem(preserveAspectRatio=false)),
+    Diagram(coordinateSystem(preserveAspectRatio=false)),
+    experiment(StopTime=8, __Dymola_Algorithm="Dassl"),
```

Comment:
I don't think that you can simulate a function so these simulation annotations can be removed?